cscs-checks/analytics/hpcg/hpcg_benchmark.py

```diff
@@ -0,0 +1,109 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+@rfm.simple_test
+class HPCGCheckRef(rfm.RegressionTest):
+    def __init__(self, **kwargs):
+        super().__init__()
+
+        self.descr = 'HPCG reference benchmark'
+        self.valid_systems = ['daint:mc', 'daint:gpu', 'dom:gpu', 'dom:mc']
+        self.valid_prog_environs = ['PrgEnv-gnu']
+        self.modules = ['craype-hugepages8M']
+        self.build_system = 'Make'
+        self.build_system.options = ['arch=MPI_GCC_OMP']
+        self.sourcesdir = None
+        self.prebuild_cmd = ['git clone https://github.com/hpcg-benchmark/hpcg.git', 'cd hpcg']
+
+        self.executable = 'hpcg/bin/xhpcg'
+        self.executable_opts = ['--nx=104', '--ny=104', '--nz=104', '-t2']
+        output_file = sn.getitem(sn.glob('HPCG*.txt'), 0)
+        self.sanity_patterns = sn.assert_eq(4, sn.count(
+            sn.findall(r'PASSED', output_file)))
+        self.num_tasks = 12
+        self.num_tasks_per_node = 12
+        self.num_cpus_per_task = 1
+
+        self.reference = {
+            'daint:gpu': {
+                'perf': (7.6, -0.1, 0.1)
+            },
+            'daint:mc': {
+                'perf': (13.4, -0.1, 0.1)
+            },
+            'dom:gpu': {
+                'perf': (7.6, -0.1, 0.1)
+            },
+            'dom:mc': {
+                'perf': (13.4, -0.1, 0.1)
+            },
+        }
+
+        self.perf_patterns = {
+            'perf': sn.extractsingle(
+                r'HPCG result is VALID with a GFLOP\/s rating of=\s*'
+                r'(?P<perf>\S+)', output_file, 'perf',  float)
+        }
+        self.maintainers = ['SK']
+
+@rfm.simple_test
+class HPCGCheckMKL(rfm.RegressionTest):
+    def __init__(self, **kwargs):
+        super().__init__()
+
+        self.descr = 'HPCG benchmark Intel MKL implementation'
+        self.valid_systems = ['dom:mc']
```

Comment:
Did you consider running this on daint also? and on the gpu partitions?

Reply:
daint has the issue with hanging flexible tests. dom:gpu would be possible if you think it makes sense to test there

Reply:
ok on daint:mc
the benchmark obviously runs on the GPU partitions as well. If it makes sense to test
cpu-only conjugate gradient there is another question and I'll leave that to you. Yes or No on daint/dom:gpu?

Reply:
@sebkelle1, I would vote for yes. Since this test can also be used for node diagnostic.

Reply:
@sebkelle1 Yes, it would make sense, cos they are different microarchitectures: Haswell @ gpu, Broadwell @ mc.